File: app/src/main/java/com/harry/myapplication/rxjava/WanAndroidApi.java
package com.harry.myapplication.rxjava;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created on 2021/9/27.
 *
 * @author harry
 */
public interface WanAndroidApi {

    @GET("project/tree/json")
    Observable<ProjectBean> getProject();

    @GET("project/list/{pageIndex}/json")
//project/list/1/json?cid=294
    Observable<ItemBean> getProjectItem( @Path("pageIndex") int pageIndex, @Query("cid") int cid);

}

File: kotlin/src/main/java/com/harry/kotlin/api/WanAndroidAPI.kt
package com.harry.kotlin.api

import com.harry.kotlin.entity.LoginRegisterResponse
import com.harry.kotlin.entity.LoginRegisterResponseWrapper
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

/**
 * Created on 2021/11/12.
 * @author harry
 */

// 对应服务器的API
interface WanAndroidAPI {

    // 登录

    @POST("/user/login")
    @FormUrlEncoded
    fun loginAction(@Field("username") username: String, @Field("password") password: String)
            : Observable<LoginRegisterResponseWrapper<LoginRegisterResponse>>


    // 注册
    @POST("/user/register")
    @FormUrlEncoded
    fun registerAction(
        @Field("username") username: String,
        @Field("password") password: String,
        @Field("repassword") repassword: String
    ): Observable<LoginRegisterResponseWrapper<LoginRegisterResponse>>
}